hyperspy/learn/mva.py

```diff
@@ -514,7 +521,7 @@ def blind_source_separation(self,
         comp_list : boolen numpy array
             choose the components to use by the boolean list. It permits
             to choose non contiguous components.
-        mask : bool numpy array or Signal instance.
+        mask : :py:class:`~hyperspy.signal.BaseSignal` (or subclass)
```

Comment:
Shall we do the same (allow only signal and not numpy array as mask) for decomposition?

Reply:
Probably yes because it simplifies the code what protect us from some bugs. However, I would leave that for another PR.